validator/lib/readTileset.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+var fs = require('fs-extra');
+var Promise = require('bluebird');
+var zlib = require('zlib');
+var isGzipped = require('./isGzipped');
+
+var fsReadFile = Promise.promisify(fs.readFile);
+var zlibGunzip = Promise.promisify(zlib.gunzip);
+
+/**
+ * Reads tileset data from a file.
+ *
+ * @param {String} filePath The file path to read from.
+ * @returns {JSON} A JSON that contains the parsed data in the filePath.
```

Comment:
This function return a promise, not JSON. Check out the description in `readTile`.